src/main/golem/util/validation/bounds.kt

```diff
@@ -0,0 +1,49 @@
+@file:JvmName("BoundsValidation")
+
+package golem.util.validation
+import golem.validateMatrices
+import golem.*
+
+/**
+ * Maximum value for individual elements in the matrix.
+ */
+var ValidationContext.max : Double
+    get() { throw UnsupportedOperationException() }
+    set(value) { max(value) }
+
+/**
+ * Ensure all of the elements in the current matrix are >= the given value.
+ * @param value The maximum value required of all elements in the matrix.
+ */
+fun ValidationContext.max(value: Double) : ValidationContext {
+    if (!validateMatrices) return this
+    currentMatrix.eachIndexed { row, col, element ->
+        if (element > value) {
+            val msg = "${currentMatrixName}[${row}, ${col}] > ${value} (value was ${element})"
+            throw IllegalArgumentException(msg)
+        }
+    }
+    return this
+}
+
+/**
+ * Minimum value for individual elements in the matrix.
+ */
+var ValidationContext.min : Double
```

Comment:
I'm not seeing the difference between min and max, but "validate by lambda" is a good enough idea to merit a rewrite anyway.